From: Ineiev <ineiev@gnu.org>
Date: Fri, 8 Apr 2016 15:17:59 +0300
Subject: tty: Refactor usage of tolower.

* tty/pinentry-tty.c (button): Apply tolower to the result.
* tty/pinentry-tty.c (confirm): Compare lowercased character instead of
converting them every time.

--

gniibe changed the original patch to keep output to TTY.
---
 tty/pinentry-tty.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/tty/pinentry-tty.c b/tty/pinentry-tty.c
index 4e71edf..d36a667 100644
--- a/tty/pinentry-tty.c
+++ b/tty/pinentry-tty.c
@@ -121,7 +121,7 @@ button (char *text, char *default_text, FILE *ttyfo)
     }
   fputc ('\n', ttyfo);
 
-  return *highlight;
+  return tolower (*highlight);
 }
 
 static void
@@ -223,17 +223,18 @@ confirm (pinentry_t pinentry, FILE *ttyfi, FILE *ttyfo)
         {
           fputc ('[', ttyfo);
           if (ok)
-            fputc (tolower (ok), ttyfo);
+            fputc (ok, ttyfo);
           if (cancel)
-            fputc (tolower (cancel), ttyfo);
+            fputc (cancel, ttyfo);
           if (notok)
-            fputc (tolower (notok), ttyfo);
+            fputc (notok, ttyfo);
           fputs("]? ", ttyfo);
         }
       fflush (ttyfo);
 
       input = fgetc (ttyfi);
       fprintf (ttyfo, "%c\n", input);
+      input = tolower (input);
 
       if (input == EOF || input == 0x4)
 	/* End of file or control-d (= end of file).  */
@@ -251,18 +252,18 @@ confirm (pinentry_t pinentry, FILE *ttyfi, FILE *ttyfo)
 	  break;
 	}
 
-      if (cancel && (input == toupper (cancel) || input == tolower (cancel)))
+      if (cancel && input == cancel)
 	{
 	  pinentry->canceled = 1;
 	  ret = 0;
 	  break;
 	}
-      else if (notok && (input == toupper (notok) || input == tolower (notok)))
+      else if (notok && input == notok)
 	{
 	  ret = 0;
 	  break;
 	}
-      else if (ok && (input == toupper (ok) || input == tolower (ok)))
+      else if (ok && input == ok)
 	{
 	  ret = 1;
 	  break;
